omnigibson/object_states/link_based_state_mixin.py

```diff
@@ -9,31 +10,49 @@ class LinkBasedStateMixin:
     def __init__(self):
         super().__init__()
 
-        self.link = None
-
-    @staticmethod
-    def get_state_link_name():
-        raise ValueError("LinkBasedState child should specify link name by overriding get_state_link_name.")
+        self._link = None
+
+    @classproperty
+    def metalink_prefix(cls):
+        """
+        Returns:
+            str: Unique keyword that defines the metalink associated with this object state
+        """
+        NotImplementedError()
+
+    @property
+    def link(self):
+        """
+        Returns:
+            None or RigidPrim: The link associated with this link-based state, if it exists
+        """
+        return self._default_link if self._link is None else self._link
+
+    @property
+    def _default_link(self):
+        """
+        Returns:
+            None or RigidPrim: If supported, the fallback link associated with this link-based state if
+                no valid metalink is found
+        """
+        # No default link by default
+        # TODO: Make this None to enforce that downstream object states have properly annotated objects
+        return self.obj.root_link
 
     def initialize_link_mixin(self):
         assert not self._initialized
 
-        self.link = None
-
-        try:
-            self.link = self.obj.links[self.get_state_link_name()]
-        except KeyError:
-            # Metalink not found, so we failed to initialize, so we assume this is a "dead" linkbasedstatemixin
-            log.warning(f"Warning: failed to initialize LinkBasedStateMixin {self.__class__.__name__} for object {self.obj.name}, no metalink"
-                            f"with name {self.get_state_link_name()} found!")
-            return False
-
-        return True
+        self._link = None
+        # TODO: Extend logic to account for multiple instances of the same metalink? e.g: _0, _1, ... suffixes
+        for name, link in self.obj.links.items():
+            if self.metalink_prefix in name:
+                self._link = link
+                break
 
-    def get_link_position(self):
-        # The necessary link is not found
+        # Raise an error if we did not find a valid link
+        # Note that we check the public accessor for self.link because a subclass might implement a fallback
+        # default_link to use
         if self.link is None:
-            return None
-
-        pos, _ = self.link.get_position_orientation()
-        return np.array(pos)
\ No newline at end of file
+            raise ValueError(f"Error: failed to initialize LinkBasedStateMixin {self.__class__.__name__} for object "
+                             f"{self.obj.name}; no metalink with prefix {self.metalink_prefix} found!")
+        return True
```

Comment:
No need to return True if it never returns False?

Reply:
done!